src/test-explorer/resolver.ts

```diff
@@ -39,10 +39,78 @@ export class TestResolver implements OnModuleInit, vscode.Disposable {
 
   onModuleInit() {
     this.ctx.subscriptions.push(this)
+    this.restorePriorTests()
+      .catch(e => {
+        this.outputChannel.appendLine(`Error restoring prior tests: ${e}`)
+      })
+      .finally(() => {
+        this.registerHandlers()
+      })
+  }
+
+  registerHandlers() {
     this.store.testController.resolveHandler = this.resolveHandler.bind(this)
     this.store.testController.refreshHandler = this.refreshHandler.bind(this)
   }
 
+  private async restorePriorTests() {
+    await this.resolveRoot()
+    const cachedBuildTargetsResult = this.store.getCachedBuildTargetsResult()
+
+    // Restore the root test item and build targets.
+    let addedTargets = false
+    if (cachedBuildTargetsResult) {
+      this.outputChannel.appendLine('Restoring prior test cases')
+      const root = this.store.testController.items.get('root')
+      if (root) {
+        try {
+          this.outputChannel.appendLine('Restoring Source Files for Root')
+          await this.processWorkspaceBuildTargetsResult(
+            root,
+            cachedBuildTargetsResult
+          )
+          root.canResolveChildren = false
+          addedTargets = true
+        } catch (e) {
+          this.outputChannel.appendLine(
+            `Error restoring source files for root: ${e}`
+          )
+        }
+      }
+    }
+
+    // Restore cached source files for targets that have them.
+    const promises: Promise<void>[] = []
+    for (const [key, testItem] of this.store.targetIdentifiers.entries()) {
+      const target: bsp.BuildTargetIdentifier = JSON.parse(key)
```

Comment:
We should try-catch-continue JSON.parse. One bad json and all the `targetIdentifiers.entries()` will fail to deserialize.